packages/medusa/src/services/discount.ts

```diff
@@ -277,36 +278,45 @@ class DiscountService extends TransactionBaseService {
   }
 
   /**
-   * Gets a discount by discount code.
-   * @param {string} discountCode - discount code of discount to retrieve
-   * @param {Object} config - the config object containing query settings
-   * @return {Promise<Discount>} the discount document
+   * Gets one or multiple discounts by discount code(s).
+   * @param discountCode - discount code(s) of discount(s) to retrieve
+   * @param config - the config object containing query settings
+   * @return the discount(s) document
    */
-  async retrieveByCode(
-    discountCode: string,
-    config: FindConfig<Discount> = {}
-  ): Promise<Discount> {
+  async retrieveByCode<
+    T extends string | string[] = string | string[],
+    TResult = T extends string[] ? Discount[] : Discount
+  >(discountCode: T, config: FindConfig<Discount> = {}): Promise<TResult> {
```

Comment:
thought: I like the idea, but discount codes are unique, so I don't see the use for it. Also, wouldn't be more reasonable to separate these into `retrieveBy*` and `listBy*` where applicable.

Reply:
I can separate it into listByCode instead indeed, make more sense